notification.c

```diff
@@ -364,21 +383,9 @@ void notification_execute_binding(struct mako_notification *notif,
 	case MAKO_BINDING_DISMISS_ALL:
 		close_all_notifications(notif->state, MAKO_NOTIFICATION_CLOSE_DISMISSED);
 		break;
-	case MAKO_BINDING_INVOKE_DEFAULT_ACTION:;
-		struct mako_action *action;
-		wl_list_for_each(action, &notif->actions, link) {
-			if (strcmp(action->key, DEFAULT_ACTION_KEY) == 0) {
-				char *activation_token = NULL;
-				if (ctx != NULL) {
-					activation_token = create_xdg_activation_token(ctx->surface,
-						ctx->seat, ctx->serial);
-				}
-				notify_action_invoked(action, activation_token);
-				free(activation_token);
-				break;
-			}
-		}
-		close_notification(notif, MAKO_NOTIFICATION_CLOSE_DISMISSED);
+	case MAKO_BINDING_INVOKE_ACTION:
+		assert(binding->action_name != NULL);
+		try_invoke_action(notif, binding->action_name, ctx);
```

Comment:
As there is only a single user now, should this keep being a function call or should I move `try_invoke_action()` back into the switch?

Reply:
I think having a separate function is worthwhile, because of the indentation level.